scripts/build.ps1

```diff
@@ -219,15 +219,26 @@ function Publish-Package
     Write-Log "Package: Publish testhost.x86\testhost.x86.csproj"
     Publish-PackageInternal $testHostx86Project $TPB_TargetFramework $testhostFullPackageDir
 
-    # Copy over the Full CLR built testhost package assemblies to the $fullCLRPackageDir
-    Copy-Item $testhostFullPackageDir\* $fullCLRPackageDir -Force
+    # Binding redirection for Newtonsoft.Json.dll has been removed from testhost config file to resolve bug: https://github.com/Microsoft/vstest/issues/391
+    # Microsoft.TestPlatform.CommunicationUtilities depends on Newtonsoft.Json of version 8.0.3 and Microsoft.TestPlatform.CrossPlatEngine depends on Microsoft.Extensions.Dependencymodel which
+    # depends on Newtonsoft.Json of version 9.0.1. So at the end testhost has Newtonsoft.Json.dll of version 9.0.1 in their publish package.
+    # Since testhost uses CommunicationUtilities (and dont use that part of code of CrossPlatEngine which uses Newtonsoft.Json of version 9.0.1) so it needs Newtonsoft.Json.dll of version 8.0.3
```

Comment:
Please also state that we have a very thin line on cross plat engine's dependency on newtonsoft.json 9.0.1(It actually should be just calling into 8.0.3 compatible APIs since that's what it finds at runtime.)